File: stacks/java-spring/backend/src/main/java/dev/brewnet/app/controller/ApiController.java
package dev.brewnet.app.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.web.bind.annotation.*;

import java.time.Instant;
import java.util.LinkedHashMap;
import java.util.Map;

@RestController
public class ApiController {

    @Autowired(required = false)
    private JdbcTemplate jdbcTemplate;

    @GetMapping("/")
    public Map<String, Object> root() {
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("service", "spring-backend");
        response.put("status", "running");
        response.put("message", "Hello Brewnet (https://www.brewnet.dev)");
        return response;
    }

    @GetMapping("/health")
    public Map<String, Object> health() {
        boolean dbConnected = false;
        if (jdbcTemplate != null) {
            try {
                jdbcTemplate.queryForObject("SELECT 1", Integer.class);
                dbConnected = true;
            } catch (Exception e) {
                // DB not available
            }
        }
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("status", "ok");
        response.put("timestamp", Instant.now().toString());
        response.put("db_connected", dbConnected);
        return response;
    }

    @GetMapping("/api/hello")
    public Map<String, Object> hello() {
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("message", "Hello from Spring Framework!");
        response.put("lang", "java");
        response.put("version", Runtime.version().toString());
        return response;
    }

    @PostMapping("/api/echo")
    public Map<String, Object> echo(@RequestBody(required = false) Map<String, Object> body) {
        return body != null ? body : Map.of();
    }
}

File: backend/src/main/kotlin/dev/brewnet/app/controller/ApiController.kt
package dev.brewnet.app.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.web.bind.annotation.*
import java.time.Instant

@RestController
class ApiController {

    @Autowired(required = false)
    private var jdbcTemplate: JdbcTemplate? = null

    @GetMapping("/")
    fun root(): Map<String, Any> = linkedMapOf(
        "service" to "springboot-kt-backend",
        "status" to "running",
        "message" to "\uD83C\uDF7A Brewnet says hello!"
    )

    @GetMapping("/health")
    fun health(): Map<String, Any> {
        val dbConnected = try {
            jdbcTemplate?.queryForObject("SELECT 1", Int::class.java)
            true
        } catch (e: Exception) {
            false
        }
        return linkedMapOf(
            "status" to "ok",
            "timestamp" to Instant.now().toString(),
            "db_connected" to dbConnected
        )
    }

    @GetMapping("/api/hello")
    fun hello(): Map<String, Any> = linkedMapOf(
        "message" to "Hello from Spring Boot (Kotlin)!",
        "lang" to "kotlin",
        "version" to KotlinVersion.CURRENT.toString()
    )

    @PostMapping("/api/echo")
    fun echo(@RequestBody(required = false) body: Map<String, Any>?): Map<String, Any> =
        body ?: emptyMap()
}
